test/ptp/ptp.go

```diff
@@ -305,6 +247,133 @@ var _ = Describe("[ptp]", func() {
 
 })
 
+func ptpConfiguration() {
+	ptpconfigList, err := client.Client.PtpConfigs(PtpLinuxDaemonNamespace).List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, ptpConfig := range ptpconfigList.Items {
+		if ptpConfig.Name == PtpGrandMasterPolicyName || ptpConfig.Name == PtpSlavePolicyName {
+			err = client.Client.PtpConfigs(PtpLinuxDaemonNamespace).Delete(ptpConfig.Name, &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+	}
+
+	nodeList, err := client.Client.Nodes().List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", PtpGrandmasterNodeLabel)})
+	Expect(err).ToNot(HaveOccurred())
+	for _, node := range nodeList.Items {
+		delete(node.Labels, PtpGrandmasterNodeLabel)
+		_, err = client.Client.Nodes().Update(&node)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	nodeList, err = client.Client.Nodes().List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", PtpSlaveNodeLabel)})
+	Expect(err).ToNot(HaveOccurred())
+	for _, node := range nodeList.Items {
+		delete(node.Labels, PtpSlaveNodeLabel)
+		_, err = client.Client.Nodes().Update(&node)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	ptpNodes, err := nodes.GetNodeTopology(client.Client)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(len(ptpNodes)).To(BeNumerically(">", 1), "need at least two nodes with ptp capable nics")
+
+	By("Labeling the grandmaster node")
+	ptpGrandMasterNode := ptpNodes[0]
+	ptpGrandMasterNode.NodeObject, err = nodes.LabelNode(ptpGrandMasterNode.NodeName, PtpGrandmasterNodeLabel, "")
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Labeling the slave node")
+	ptpSlaveNode := ptpNodes[1]
+	ptpSlaveNode.NodeObject, err = nodes.LabelNode(ptpSlaveNode.NodeName, PtpSlaveNodeLabel, "")
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Creating the policy for the grandmaster node")
+	err = createConfig(PtpGrandMasterPolicyName,
+		ptpGrandMasterNode.InterfaceList[0],
+		"",
+		"-a -r -r",
+		PtpGrandmasterNodeLabel,
+		pointer.Int64Ptr(5))
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Creating the policy for the slave node")
+	err = createConfig(PtpSlavePolicyName,
+		ptpSlaveNode.InterfaceList[0],
+		"-s",
+		"-a -r",
+		PtpSlaveNodeLabel,
+		pointer.Int64Ptr(5))
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Restart the linuxptp-daemon pods")
+	ptpPods, err := client.Client.Pods(PtpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+	Expect(err).ToNot(HaveOccurred())
+	for _, pod := range ptpPods.Items {
+		err = client.Client.Pods(PtpLinuxDaemonNamespace).Delete(pod.Name, &metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
+		Expect(err).ToNot(HaveOccurred())
+	}
+}
+
+// Returns the slave node label to be used in the test
+func checkPolicies(namespace string) (string, string, string, string) {
```

Comment:
I'd rename this to `discoveryPTPConfiguration`.
Also, maybe packing the returned strings into a struct is more readable